tools/p4mlir-export/simplebmv2.p4

```diff
@@ -0,0 +1,69 @@
+// Simplest p4 program to be processed, which can go through
```

Comment:
Why is the mlir code for this file empty? Does it not compile? What is the expected json output for it? Maybe you should add it as a reference.

Reply:
oh, forgot to upload. let me upload it

Reply:
Try this code as example: https://github.com/p4lang/p4c/blob/main/backends/p4tools/modules/testgen/targets/bmv2/test/p4-programs/bmv2_extract_1.p4

I know this already compiles to mlir and it is a very simple example of a BMV2 program. Then compile this program and generate the json file using p4c-bm2-ss. You should be able to generate basic elements already. 

I am not sure what looped references or typedefs are.

Reply:
How to produce text output from this particular MLIR dialect you can check https://github.com/p4lang/p4mlir-incubator/blob/fruffy/top4/lib/Utilities/mlir_to_p4.h#L20. It writes MLIR to P4 again. This code is WiP, by no means complete, and will change. But it is a start and shows that it can be done.

Reply:
> I am not sure what looped references or typedefs are.
For example:
```
!b16i = !p4hir.bit<16>
```
After the p4mlir-translate, the code gets expanded, and also linked with other v1 arch codes.
for loop reference, I mean, it is dealing with my traverse methods, it is not a tree, could have nodes being traversed.
My current code hangs at complex p4hir, goes into an infinite loop. Lastly, the loop syntax they currently support would not be easily represented as a JSON DAG if it is not deterministic.

Sure, thx, I will check your code.

Reply:
Don't worry about for loops for now, there is very little Bmv2 code written that way. Just throw an error that it is not implemented,

Reply:
> Don't worry about for loops for now, there is very little Bmv2 code written that way. Just throw an error that it is not implemented,

got you, thx

Reply:
@Joejiong Could you also check in the expected BMv2 `.json` file? I still don't see it.

Reply:
OK, let me include it

Reply:
> @Joejiong Could you also check in the expected BMv2 `.json` file? I still don't see it.

The output also has a p4i file, I probably only upload the `.json`

Reply:
Yes, including `.json` file is enough.